src/cli_v1.rs

```diff
@@ -176,6 +192,10 @@ pub fn run_cli_v1_main() -> anyhow::Result<()> {
         }
     }
 
+    if !args.no_exit_1_on_error && error_occurred.load(Ordering::SeqCst) {
+        std::process::exit(1);
```

Comment:
instead of exiting 1 immediately (which skips destructors and might be otherwise weird), you can just return an `Err`; something like this
```suggestion
        anyhow::bail!("an error was logged");
```